NEWS.md

```diff
@@ -22,6 +22,8 @@
 
 4. The translations submitted for 1.16.0 are now actually shipped with the package -- our deepest apologies to the translators for the omission. We have added a CI check to ensure that the .mo binaries which get shipped with the package are always up-to-date.
 
+5. Grouping operations now use internal `wallclock()` for verbose reporting, reducing a significant overhead on non-Windows platforms introduced by C's native `clock()` function.
```

Comment:
Generally I think we should avoid writing about highly internal things like "wallclock" in the NEWS

```suggestion
5. Some grouping operations run much faster under `verbose=TRUE`, [#6286](https://github.com/Rdatatable/data.table/issues/6286). Thanks @joshhwuu for the report and fix. This overhead was not present on Windows. As a rule, users should expect `verbose=TRUE` operations to run more slowly, as extra statistics might be calculated as part of the report; here was a case where the overhead was particularly high and the fix was particularly easy.
```

Reply:
I agree, abstraction might be better for such details

Reply:
Also, nice edit!